common/src/auth/ServiceConnectionStringParser.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.DigitalTwin.Model.Service;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// The Service Connection String Parser class
+    /// </summary>
+    public class ServiceConnectionStringParser
+    {
+        protected const char ValuePairDelimiter = ';';
+        protected const char ValuePairSeparator = '=';
+        protected const string HostNameSeparator = ".";
+
+        protected string HostNamePropertyName = "HostName";
+        protected string SharedAccessKeyNamePropertyName = "SharedAccessKeyName";
+        protected string SharedAccessKeyPropertyName = "SharedAccessKey";
+        protected string SharedAccessSignaturePropertyName = "SharedAccessSignature";
+
+        protected static TimeSpan regexTimeoutMilliseconds = TimeSpan.FromMilliseconds(500);
```

Comment:
If this will not change, mark it readonly? If so, also case it like it is const.

Reply:
Regex does not take non static field so not possible to mark this as readonly

Reply:
No, I mean static and readonly

Reply:
got it

Reply:
Still not static and readonly